packages/mui-joy/src/AccordionDetails/AccordionDetails.test.tsx

```diff
@@ -29,107 +29,69 @@ describe('<AccordionDetails />', () => {
     },
   }));
 
-  describe('tab index', () => {
-    it('[initial] interactive content should have tab index -1', () => {
+  describe('hidden attribute', () => {
```

Comment:
Ideally, tests shouldn't check for the `hidden` attribute being present or not. I consider that an implementation detail. Instead, we should try to test if the accordion details or its contents are reachable or not. On the topic https://testing-library.com/docs/guide-disappearance/

```suggestion
  describe('expand/collapse', () => {
```

To be honest, feels like testing these `AccordionDetails` behaviors should be tested as part of `Accordion` and not in isolation, but let's leave that for another PR.